ultralytics/data/explorer/explorer.py

```diff
@@ -53,12 +55,15 @@ def build_transforms(self, hyp=None):
 
 class Explorer:
 
-    def __init__(self, data='coco128.yaml', model='yolov8n.pt', uri='~/ultralytics/explorer') -> None:
+    def __init__(self,
+                 data: Union[str, Path] = 'coco128.yaml',
+                 model: str = 'yolov8n.pt',
+                 uri: str = '~/ultralytics/explorer') -> None:
         checks.check_requirements(['lancedb', 'duckdb'])
         import lancedb
 
         self.connection = lancedb.connect(uri)
-        self.table_name = Path(data).name.lower() + '_' + model.lower()
+        self.table_name = Path(data).stem.lower() + '_' + Path(model.lower()).stem
```

Comment:
@Burhan-Q this was done for a reason to avoid name collision. For example, there can be a dataset with same name without .yaml (think classification).. The more specific we get the chance of name collision goes down

Reply:
Leaving the `suffix` threw an error when running a SQL query. I'll change back and post the exact error, but it essentially boiled down to "no table `pt`" since the SQL query is going to parse `parent.child` as a full address based on hierarchy.

Reply:
What do you think about using `(Path(data).name.lower() + "_" + Path(model.lower()).name).replace(".","_")` to avoid the use of `.` in the table name?

Reply:
@Burhan-Q Can you please share an example of sql query you're running. It should not require table name, you can simply run `SELECT * FROM 'table'`

Reply:
@AyushExel

```py
from pathlib import Path
from ultralytics import Explorer

exp = Explorer(data=r"Q:\datasets\coco128\coco128.yaml", model="yolov8n.pt")
exp.create_embeddings_table()
query = 'SELECT * FROM table WHERE labels LIKE "%person%"'
result = exp.plot_sql_query(query)
```

## Error

```
Running query: SELECT * FROM table WHERE labels LIKE "%person%"

Traceback (most recent call last):

  File "<stdin>", line 1, in <module>
  File "Q:\ML_dev\ultra_repo\ultralytics-1\ultralytics\data\explorer\explorer.py", line 229, in plot_sql_query
    result = self.sql_query(query, return_type='arrow')
  File "Q:\ML_dev\ultra_repo\ultralytics-1\ultralytics\data\explorer\explorer.py", line 205, in sql_query
    rs = duckdb.sql(query)

duckdb.duckdb.ParserException: Parser Error: syntax error at or near "table"

LINE 1: SELECT * FROM table WHERE labels LIKE "%person%"
                      ^
```

Reply:
I copied the `query` line directly from the doc string for `plot_sql_query`

Reply:
Also tried the same for `plot_sql_query`

```py
query = 'SELECT * FROM table WHERE labels LIKE "%person%"'
exp.plot_sql_query(query)
```

## Error 
```
Running query: SELECT * FROM table WHERE labels LIKE "%person%"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "Q:\ML_dev\ultra_repo\ultralytics-1\ultralytics\data\explorer\explorer.py", line 229, in plot_sql_query
    result = self.sql_query(query, return_type='arrow')
  File "Q:\ML_dev\ultra_repo\ultralytics-1\ultralytics\data\explorer\explorer.py", line 205, in sql_query     
    rs = duckdb.sql(query)
duckdb.duckdb.ParserException: Parser Error: syntax error at or near "table"
LINE 1: SELECT * FROM table WHERE labels LIKE "%person%"
                      ^
```

Reply:
@AyushExel just saw your recommendation and yes, this works!

```py
exp.sql_query("SELECT * FROM 'table'")

0    Q:\datasets\coco128\images\train\000000000009.jpg  [bowl, bowl, bowl, orange, orange, orange, ora...  ...    [[[]]]  [0.037930287420749664, -0.14796601235866547, -...       
1    Q:\datasets\coco128\images\train\000000000025.jpg                                 [giraffe, giraffe]  ...    [[[]]]  [-0.14058472216129303, -0.1475253850221634, -0...       
2    Q:\datasets\coco128\images\train\000000000030.jpg                               [potted plant, vase]  ...    [[[]]]  [-0.05795533210039139, -0.0636223629117012, -0...       
3    Q:\datasets\coco128\images\train\000000000034.jpg                                            [zebra]  ...    [[[]]]  [-0.17022110521793365, -0.14574818313121796, -...       
4    Q:\datasets\coco128\images\train\000000000036.jpg                                 [person, umbrella]  ...    [[[]]]  [-0.1257043331861496, -0.19695159792900085, -0...       
..                                                 ...                                                ...  ...       ...                                                ...       
110  Q:\datasets\coco128\images\train\000000000584.jpg  [spoon, bowl, broccoli, broccoli, carrot, carr...  ...    [[[]]]  [0.009227197617292404, -0.15530605614185333, -...       
111  Q:\datasets\coco128\images\train\000000000589.jpg                                  [person, frisbee]  ...    [[[]]]  [-0.16230307519435883, -0.20778605341911316, -...       
112  Q:\datasets\coco128\images\train\000000000590.jpg                                       [bowl, sink]  ...    [[[]]]  [-0.09467365592718124, -0.19775943458080292, -...       
113  Q:\datasets\coco128\images\train\000000000595.jpg                                               [tv]  ...    [[[]]]  [-0.07988807559013367, -0.23142163455486298, -...       
114  Q:\datasets\coco128\images\train\000000000597.jpg  [elephant, elephant, elephant, elephant, eleph...  ...    [[[]]]  [-0.1691778004169464, -0.07819393277168274, -0...       

[115 rows x 7 columns]
```

Reply:
```py
 exp.sql_query("SELECT * FROM 'table' WHERE labels LIKE '%person%'")
 ```
 
 Also works and I updated the doc string to use this.

Reply:
Perfect.. @Burhan-Q you can even eliminate the SELECT * FROM 'table' part..and simply start from the WHERE clause. 
Thanks for updating the docs